doc/make.jl

```diff
@@ -42,6 +43,62 @@ cd(joinpath(@__DIR__, "src")) do
+# of makedocs(), to make sure that Documenter knows how to resolve the directories
+# in stdlib/ to the correct remote Git repositories (for source and edit links).
+#
+# This function parses the *.version files in stdlib/, returning a dictionary with
+# all the key-value pairs from those files. *_GIT_URL and *_SHA1 fields are the ones
+# we will actually be interested in.
+function parse_stdlib_version_file(path)
+    values = Dict{String,String}()
+    for line in readlines(path)
+        m = match(r"^([A-Z0-9_]+)\s+:?=\s+(\S+)\s?$", line)
+        if isnothing(m)
+            @warn "Unable to parse line in $(path)" line
+        else
+            values[m[1]] = m[2]
+        end
+    end
+    return values
+end
+# This generates the value that will be passed to the `remotes` argument of makedocs(),
+# by looking through all *.version files in stdlib/.
+documenter_stdlib_remotes = let stdlib_dir = realpath(joinpath(@__DIR__, "..", "stdlib"))
+    # Get a list of all *.version files in stdlib/..
+    version_files = filter(readdir(stdlib_dir)) do fname
+        isfile(joinpath(stdlib_dir, fname)) && endswith(fname, ".version")
+    end
+    # .. and then parse them, each becoming an entry for makedocs's remotes.
+    # The values for each are of the form path => (remote, sha1), where
+    #  - path: the path to the stdlib package's root directory, i.e. "stdlib/$PACKAGE"
+    #  - remote: a Documenter.Remote object, pointing to the Git repository where package is hosted
+    #  - sha1: the SHA1 of the commit that is included with the current Julia version
+    remotes_list = map(version_files) do version_fname
+        package = match(r"(.+)\.version", version_fname)[1]
+        versionfile = parse_stdlib_version_file(joinpath(stdlib_dir, version_fname))
+        # From the (all uppercase) $(package)_GIT_URL and $(package)_SHA1 fields, we'll determine
+        # the necessary information. If this logic happens to fail for some reason for any of the
+        # standard libraries, we'll crash the documentation build, so that it could be fixed.
+        remote = let git_url_key = "$(uppercase(package))_GIT_URL"
+            haskey(versionfile, git_url_key) || error("Missing $(git_url_key) in $version_fname")
+            m = match(LibGit2.GITHUB_REGEX, versionfile[git_url_key])
+            isnothing(m) && error("Unable to parse $(git_url_key)='$(versionfile[git_url_key])' in $version_fname")
+            Documenter.Remotes.GitHub(m[2], m[3])
+        end
+        package_sha = let sha_key = "$(uppercase(package))_SHA1"
+            haskey(versionfile, sha_key) || error("Missing $(sha_key) in $version_fname")
+            versionfile[sha_key]
+        end
+        # Construct the absolute (local) path to the stdlib package's root directory
+        package_root_dir = joinpath(stdlib_dir, "$(package)-$(package_sha)")
+        isdir(package_root_dir) || error("Missing stdlib: $(package_root_dir)")
```

Comment:
The second issue, that I initially missed, is that we can't find the stdlib:

```
ERROR: LoadError: Missing stdlib: /cache/build/default-amdci5-4/julialang/julia-master/stdlib/ArgTools-4eccde45ddc27e4f7fc9094b2861c684e062adb2
```

https://buildkite.com/julialang/julia-master/builds/27823#018aa5df-3848-40d1-9c9d-0623c89bdd21/713-752

@staticfloat So the `stdlib/` directory gets cleaned out, since we start anew, and just extract `usr`? Locally, `usr/share/julia/stdlib/v1.11/` are just symlinks to `stdlib/`. But the `.version` files are missing as far as I can tell.

Note: all this works locally

Reply:
You're working in a build tree; but this is not a build tree, it's the result of `make install` into `usr`.

Reply:
What's the best way to replicate the CI worktree locally? Should I just `cp -R` copy `usr/` into a clean Git worktree? The symlinks would still be there though?

It's a bit unfortunate that we're not building the docs in the build tree, to match what happens locally. But if there is a way to access the stdlib info consistently in both cases, then that should be fine.

Reply:
There are no symlinks, because this is the result of `make install`, not just `make`.

You can take a clean `git` checkout, and then download the same tarball that is built from buildkite; this one for instance: https://buildkite.com/organizations/julialang/pipelines/julia-master/builds/27823/jobs/018aa5df-4d2c-41ad-a0be-226f905cdd6e/artifacts/018aa5e8-571c-41a6-9cb6-656215210e84